duckduckgo/macos-browser
# Add dedicated subscription target

Task/Issue URL: https://app.asana.com/0/0/1205619843916174/f
Tech Design URL:
CC: @miasma13 

**Description**:

This PR adds a new `DuckDuckGo Privacy Pro` target which contains the subscription feature.

This PR also updates `archive.sh` to support building the new target.

**Steps to test this PR**:
1. Check that the main `DuckDuckGo Privacy Browser` target builds, and that it has no trace of the Privacy Pro work
2. Check that the `DuckDuckGo Privacy Browser App Store` target builds, and that it has no trace of the Privacy Pro work
3. Check that the `DuckDuckGo Privacy Pro` target builds and that the subscriptions feature works, and that Network Protection and DBP are visible after enabling the subscription
4. Check that `./scripts/archive.sh subscription` works and creates a working subscriptions build (you may need to turn off SwiftLint)



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
